larpandoracontent/LArTwoDReco/LArClusterAssociation/CrossGapsExtensionAlgorithm.cc

```diff
@@ -20,21 +20,21 @@ namespace lar_content
 {
 
 CrossGapsExtensionAlgorithm::CrossGapsExtensionAlgorithm() :
-    m_minClusterLength(5.f),
-    m_minGapFraction(0.5f),
-    m_maxGapTolerance(2.f),
+    m_minClusterLength(10.f),
+    m_maxGapTolerance(2.5f),
     m_maxTransverseDisplacement(2.5f),
-    m_maxRelativeAngle(10.f)
+    m_maxRelativeAngle(15.f),
+    m_minCosRelativeAngle(0.966)
 {
 }
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
 void CrossGapsExtensionAlgorithm::GetListOfCleanClusters(const ClusterList *const pClusterList, ClusterVector &clusterVector) const
 {
-    // ATTN May want to opt-out completely if no gap information available
-    // if (PandoraContentApi::GetGeometry(*this)->GetDetectorGapList().empty())
-    //     return;
+    // ATTN - Opt out completely if there is no gap information available
+    if (PandoraContentApi::GetGeometry(*this)->GetDetectorGapList().empty())
+         return;
```

Comment:
Presumably this could change reconstruction output, although there are now few detector models which have no registered gaps (just CR steps for DUNE multi-volume detectors)? This, then, is the kind of innocuous line that will raise questions later on.

Are you sure this is the way forwards, and can vouch for it later? As an intermediate, can always make this on/off configurable. If you think it should behave in the new way, we should push to have it work in this new manner.

Reply:
This won't change any output (will it?). The two CrossGaps algorithms will only modify clusters that are in proximity to detector gaps. If there are no gaps, the algorithms will ultimately do nothing. It seems most efficient to bail out straight away if there are no gaps.

Reply:
I agree, that it will only have an impact if the algorithm actually ended up making cluster merges even if there were no detector gaps in the vicinity! I don't know whether this remains the case for our gap-crossing algorithms, but I firmly recall that they did do this immediately after implementation, back in 2016. (Basically, they contained new cluster-association mechanisms, of which the gaps played a strong role, but weren't mandatory. Not ideal, certainly not from an aesthetic point of view, but the merges all looked good when I last examined this!)